oshi-core/src/main/java/oshi/util/FileUtil.java

```diff
@@ -119,7 +124,9 @@ public static List<String> readLines(String filename, int count, boolean reportE
             if (LOG.isDebugEnabled()) {
                 LOG.debug(READING_LOG, filename);
             }
-            try (BufferedReader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
+            CharsetDecoder decoder = StandardCharsets.UTF_8.newDecoder();
```

Comment:
Are we sure the files we are reading will always be in UTF-8?

We don't currently read any files in Windows, but the util should be generic.  And while procsys files are going to be a defined (ASCII) format, we also read various `/etc/` and other files which in theory could use a different character set.

Reply:
Yes, I think that it's a bit problematic to have generic utility classes that read text files without allowing the caller to specify the character set with which to interpret the bytes, but that's where we are now, since that's what `FileUtil.readFile(String)` did previously (assumed UTF-8).

The previous PR provides an implementation that allows for reading a specific number of lines as UTF-8 strings (following roughly how `Files.readAllLines(Path, Charset)` is implemented.

This PR expands the same thing that `Files.newBufferedReader(Path, Charset)` does, but with a 1k buffer.

I think it would potentially make sense to have different types of reader classes that understand how to process specific types of files (for example a class for binary `/proc` files and a class for text `/proc` files, and a class for each different types of file in `/etc`, etc).